webapp/nuxt.config.js

```diff
@@ -106,17 +121,98 @@ module.exports = {
     middleware: ['authenticated'],
     linkActiveClass: 'router-link-active',
     linkExactActiveClass: 'router-link-exact-active',
-    scrollBehavior: () => {
-      return { x: 0, y: 0 }
+    scrollBehavior: (to, _from, savedPosition) => {
+      let position = false
+      // if no children detected and scrollToTop is not explicitly disabled
+      if (
+        to.matched.length < 2 &&
+        to.matched.every(r => r.components.default.options.scrollToTop !== false)
+      ) {
+        // scroll to the top of the page
+        position = {
+          x: 0,
+          y: 0,
+        }
+      } else if (to.matched.some(r => r.components.default.options.scrollToTop)) {
+        // if one of the children has scrollToTop option set to true
+        position = {
+          x: 0,
+          y: 0,
+        }
+      }
+
+      // savedPosition is only available for popstate navigations (back button)
+      if (savedPosition) {
+        position = savedPosition
+      }
+
+      return new Promise(resolve => {
+        // wait for the out transition to complete (if necessary)
+        window.$nuxt.$once('triggerScroll', () => {
```

Comment:
I hope this is SSR safe! (although considering it's the `nuxt.config.js` it should be the case)

Reply:
One would hope so … 😉